src/dcos_e2e/cluster.py

```diff
@@ -90,6 +92,61 @@ def from_nodes(
             cluster_backend=backend,
         )
 
+    @retry(
+        exceptions=(
+            subprocess.CalledProcessError,
+            ValueError,
+            requests.exceptions.ConnectionError,
+        ),
+        tries=500,
+        delay=5,
+    )
+    def _wait_for_adminrouter(
```

Comment:
Why AdminRouter specifically?

Reply:
Hmm, what a better name for: Wait until the CA certificate could be reached if this was an EE cluster or until Admin Router reports that the CA is not found if it is an OSS cluster.
Could call it wait_for_ca_cert_up_or_not_found maybe.

Reply:
Instead, what do you think of splitting this up.

`wait_for_dcos_diagnostics` or maybe `wait_for_systemd_units` as one function.

Then `wait_for_ca_cert` in another. Then we could just not call `wait_for_ca_cert` in the OSS helper. And maybe we would keep your existing code to wait for a CA Cert.

Reply:
Calling it _wait_for_dcos_startup now.

Reply:
I totally agree. However I wanted to keep it as close as possible first to see if it even works.

Reply:
Ok - well do you mind seeing if the tests pass and then rearranging and then resubmitting?

Reply:
Yes, that's what I was going to do. I really want to see the EE tests pass first in Enterprise